src/hm6/services/profile.service.ts

```diff
@@ -0,0 +1,87 @@
+import { Cart, Order } from "../models";
+import dataBase from "../repositories/db.repository";
+import { calculateTotal } from "./helpers";
+import { getProductDetails } from "./product.service";
+const { v4: uuidv4 } = require("uuid");
+
+export function getOrCreateUserCart(userId: string): Cart {
+  const existingCart = dataBase.carts.find(
+    (cart) => cart.userId === userId && !cart.isDeleted
+  );
+
+  if (existingCart) {
+    return existingCart;
+  }
+
+  const newCart: Cart = {
+    id: uuidv4(),
+    userId: userId,
+    isDeleted: false,
+    items: [],
+  };
+
+  dataBase.carts.push(newCart);
+
+  return newCart;
+}
+
+export function updateExistedUserCart(
+  productId: string,
+  count: number,
+  userId: string
+): Cart {
+  const userCartIndex = dataBase.carts.findIndex((c) => c.userId === userId);
```

Comment:
Won't you will be updating deleted cart if `isDeleted` will be `true`?

Or `isDeleted` property of the cart does not make sense :)